src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/PiglinBarterDrop.java

```diff
@@ -0,0 +1,39 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.event.entity.EntityDropItemEvent;
+
+import io.github.thebusybiscuit.slimefun4.implementation.items.misc.BasicCircuitBoard;
+import io.github.thebusybiscuit.slimefun4.implementation.listeners.MobDropListener;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+/**
+ * This interface, when attached to a {@link SlimefunItem}, provides an easy method for adding
+ * a % chance to drop for an {@link SlimefunItem} on {@link EntityItemDropEvent}, this chance is 0-100
+ * and used in conjunction with the BARTER_DROP {@link RecipeType}.
+ * 
+ * @author dNiym
+ *
+ * @see MobDropListener
+ * 
+ */
+@FunctionalInterface
+public interface PiglinBarterDrop extends RandomMobDrop {
```

Comment:
Also I forgot the method was called getMobDropChance(). Maybe not extend that and just make a method called getBarteringLootChance() instead?